model/repositories.go

```diff
@@ -286,6 +296,80 @@ func (e *EventRepositoryGorm) Remove(entities []Entity) error {
 	return nil
 }
 
+//Content may not be applicable to this func since there would be an instance of it being called at server.go run. Therefore we won't have a "proper" content which would contain the EntityFactory
+func (e *EventRepositoryGorm) ReplayEvents(ctxt context.Context, date time.Time, entityFactories map[string]EntityFactory, projections Projection) (int, int, int, error) {
+	schemas := make(map[string]ds.Builder)
+
+	for _, value := range entityFactories {
+		schemas[value.Name()] = value.Builder(context.Background())
+	}
+
+	err := projections.Migrate(ctxt, schemas)
+	if err != nil {
+		e.logger.Errorf("error migrating tables: %s", err)
+	}
+
+	var events []GormEvent
+
+	if date.IsZero() {
+		result := e.DB.Table("gorm_events").Find(&events)
+		if result.Error != nil {
+			e.logger.Errorf("got error pulling events '%s'", result.Error)
+			return 0, 0, 0, result.Error
+		}
+	} else {
+		result := e.DB.Table("gorm_events").Where("created_at =  ?", date).Find(&events)
+		if result.Error != nil {
+			e.logger.Errorf("got error pulling events '%s'", result.Error)
+			return 0, 0, 0, result.Error
+		}
+	}
+
+	var tEvents []*Event
+
+	for _, event := range events {
+		tEvents = append(tEvents, &Event{
+			ID:      event.ID,
+			Type:    event.Type,
+			Payload: json.RawMessage(event.Payload),
+			Meta: EventMeta{
+				EntityID:   event.EntityID,
+				EntityType: event.EntityType,
+				RootID:     event.RootID,
+				Module:     event.ApplicationID,
+				User:       event.User,
+				SequenceNo: event.SequenceNo,
+				SchemaName: event.SchemaName,
+			},
+			Version: 0,
+		})
+	}
+
+	totalEvents := len(tEvents)
+	successfulEvents := 0
+	failedEvents := 0
+	entity := map[string]interface{}{}
+
+	for _, event := range tEvents {
+
+		newContext := context.WithValue(ctxt, context2.ENTITY_FACTORY, entityFactories[event.Meta.SchemaName])
+
+		result := e.DB.Table(event.Meta.SchemaName).Find(&entity, "weos_id = ? and sequence_no = ?", event.Meta.EntityID, event.Meta.SequenceNo)
```

Comment:
I'm not sure why this is needed. I think simply dispatch the events and the projection that is associated with the event store will respond accordingly